### PR TITLE
Document virtual environment recreate rules

### LINKED ISSUE #93: Environment is not recreated when install_requires in setup.py is changed
- Bitbucket: https://bitbucket.org/hpk42/tox/issue/93
- Originally reported by: @tkf
- Originally created at: 2013-04-15T22:07:44.806

Here is a recipe to reproduce this problem.
1. Run tox in some project.
2. Add a new package to install_requires in setup.py
3. Run tox again.

I can write dependency in tox.ini to force recreate, but I think it is simpler if tox reinstall package.  I think it is as easy as removing --no-deps flag so I wonder why --no-deps is used in the first place.  Is there any drawbacks?  Wouldn't it be nice if we can disable this flag optionally?

### PR DESCRIPTION
**Pull request by gaborbernat**
Resolves #93.

https://tox.readthedocs.io/en/93/example/general.html#recreate

URL of RTD document: https://tox.readthedocs.io/en/93/

### CODE REVIEW

File: docs/example/general.rst [hunk 1]
```diff
@@ -39,36 +39,43 @@ syntax:
...
+- if it's a library project (has build package phase), install library dependencies
+  (with potential extras)
+
+These three steps are only performed once (given they all succeeded). Subsequent calls
+that don't detect changes to the traits of that step will not alter the virtual
+environment in any way. When a change is detected for any of the steps, the entire
+virtual environment is removed and the operation starts from scratch (this is
+because it's very hard to determine what would the delta changes would be needed -
+e.g. a dependency could migrate from one dependency to another, and in this case
+we would need to install the now while removing the old one).
```
Comment (nicoddemus) [inline docs/example/general.rst]: typo: now -> new

File: docs/example/general.rst [hunk 2]
```diff
@@ -39,36 +39,43 @@ syntax:
...
+
+Here's what traits we track at the moment for each steps,
```
Comment (nicoddemus) [inline docs/example/general.rst]: , -> :

File: docs/example/general.rst [hunk 3]
```diff
@@ -39,36 +39,43 @@ syntax:
...
+
+- virtual environment trait is tied to the python path the :conf:`basepython`
+  resolves too (if this config changes, the virtual environment will be recreated),
+- change inside the :conf:`deps` sections mean change at string level for the entries,
```
Comment (nicoddemus) [inline docs/example/general.rst]: I think this is a little confusing, perhaps:

> :conf:`deps` sections changes (meaning any string-level change for the entries),

**Comment by gaborbernat** [comment]
Feedback incorporated, thanks :+1: see tox.readthedocs.io/en/93/example/general.html#recreate

**Pull request merged**